src/lfo.cpp

```diff
@@ -40,6 +40,25 @@ int32_t LFO::render(int numSamples, int waveType, uint32_t phaseIncrement) {
 	case LFO_TYPE_TRIANGLE:
 		value = getTriangle(phase);
 		break;
+
+	case LFO_TYPE_SAH:
+		if (phase == 0) {
+			value = CONG;
+			holdValue = value;
+		}
+		else if (phase + phaseIncrement * numSamples < phase) holdValue = CONG;
+		else value = holdValue;
+		break;
+
+	case LFO_TYPE_RWALK:
+        uint32_t range = 4294967295u / 20;
+		if (phase == 0) {
+			value = holdValue + (range / 2) - CONG % range;
```

Comment:
For LFO2 this should reset to a constant instead of using the holdValue

Reply:
You mean that LFO should essentially always start at zero modulation? That is a little difficult to do when at the same time LFO1 sync should not return to zero at the start of every sync interval, which we can of course discuss as well. In both cases the LFO phase is reset to 0.

I think these random LFOs will always behave a little different than the original periodic ones. One example is that I made them tick half as fast, so only one value change per sync interval as opposed to two maxima in the periodic LFOs, which means two flips for square. In my opinion, that is not a bad thing as long as it is somewhat intuitive.

Interested to hear your opinion on this.

Reply:
Yeah thats a good point, I didn't consider how it would interact with LFO1. I don't have a strong opinion on whether the LFO2 version would ideally reset to zero or to a random value on with every note on but the current free running behaviour is un intuitive with how LFO2 normally works. 

I don't have a good solution since I forgot about LFO1 but I'll comment again if I think of one

Reply:
I tried to replicate what you describe, but I am not sure if I completely understand what you mean. To me, it sounds like LFO2 resets correctly on every trigger. Not to zero, but to a new random value, which would make sense with how it is implemented at the moment. The phase gets reset to zero when a new note triggers and that triggers a new LFO cycle, which means a new value for the random LFOs. That is very similar to the tempo synced behavior in LFO1, but feels a little different than for example "sine" which will always start at zero. To me, it feels somewhere in between free running (in terms of modulation) and re-triggering (in terms of rhythm).

Does that make sense? I might be completely misunderstanding your point, though.

Reply:
The S+H mode sounds like you're describing but the random walk doesn't. If you wait until it's audibly at a low point then when you hit a new note on it stays low. I think this is because the phase reset at 0 still uses the previous hold value 

It's not a huge deal and if it can't be changed without affecting LFO1 then I don't think it's a show stopper, the overall effect definitely sounds cool

Reply:
Got it now, thank you. I just pushed some changes that should address that and the biasing.